uaclient/contract.py

```diff
@@ -264,9 +264,9 @@ def get_magic_attach_token_info(self, magic_token: str) -> Dict[str, Any]:
             response = self.request_url(
                 API_V1_GET_MAGIC_ATTACH_TOKEN_INFO, headers=headers
             )
-        except exceptions.UrlError as e:
-            LOG.exception(e)
-            raise exceptions.ConnectivityError()
+        except exceptions.ConnectivityError as e:
+            LOG.exception(e.cause_error)
+            raise e
```

Comment:
There are a couple places where we catch ConnectivityError, log the message, and then re-raise it. Maybe we should log the error when raising it originally in http/__init__.py and then remove all the catch-log-reraise blocks - what do you think?

Reply:
That makes sense, I will update the PR